frontend/widgets/OBSBasic.cpp

```diff
@@ -1658,127 +1648,63 @@ bool OBSBasic::ResetAudio()
-	 * second until we've exit any known sub-loops. */
-	if (os_atomic_load_long(&insideEventLoop) != 0) {
-		QTimer::singleShot(1000, this, &OBSBasic::close);
-		event->ignore();
-		return;
-	}
+	OBSMainWindow::close();
+}
 
-#ifdef YOUTUBE_ENABLED
-	/* Also don't close the window if the youtube stream check is active */
-	if (youtubeStreamCheckThread) {
-		QTimer::singleShot(1000, this, &OBSBasic::close);
-		event->ignore();
+void OBSBasic::closeEvent(QCloseEvent *event)
+{
+	if (isClosePromptOpen || isClosing()) {
 		return;
 	}
-#endif
 
-	if (isVisible())
+	if (isVisible()) {
 		config_set_string(App()->GetUserConfig(), "BasicWindow", "geometry",
 				  saveGeometry().toBase64().constData());
+	}
 
-	bool confirmOnExit = config_get_bool(App()->GetUserConfig(), "General", "ConfirmOnExit");
-
-	if (confirmOnExit && outputHandler && outputHandler->Active() && !clearingFailed) {
-		SetShowing(true);
-
-		QMessageBox::StandardButton button =
-			OBSMessageBox::question(this, QTStr("ConfirmExit.Title"), QTStr("ConfirmExit.Text"));
+	if (!isReadyToClose()) {
+		event->ignore();
 
-		if (button == QMessageBox::No) {
-			event->ignore();
-			restart = false;
-			return;
-		}
+		QTimer::singleShot(1000, this, &OBSBasic::close);
+		return;
 	}
 
 	if (remux && !remux->close()) {
 		event->ignore();
 		restart = false;
-		return;
-	}
 
-	QWidget::closeEvent(event);
-	if (!event->isAccepted())
 		return;
-
-	blog(LOG_INFO, SHUTDOWN_SEPARATOR);
-
-	closing = true;
-
-	/* While closing, a resize event to OBSQTDisplay could be triggered.
-	 * The graphics thread on macOS dispatches a lambda function to be
-	 * executed asynchronously in the main thread. However, the display is
-	 * sometimes deleted before the lambda function is actually executed.
-	 * To avoid such a case, destroy displays earlier than others such as
-	 * deleting browser docks. */
-	ui->preview->DestroyDisplay();
-	if (program)
-		program->DestroyDisplay();
-
-	if (outputHandler->VirtualCamActive())
-		outputHandler->StopVirtualCam();
-
-	if (introCheckThread)
-		introCheckThread->wait();
-	if (whatsNewInitThread)
-		whatsNewInitThread->wait();
-	if (updateCheckThread)
-		updateCheckThread->wait();
-	if (logUploadThread)
-		logUploadThread->wait();
-	if (devicePropertiesThread && devicePropertiesThread->isRunning()) {
-		devicePropertiesThread->wait();
-		devicePropertiesThread.reset();
 	}
 
-	QApplication::sendPostedEvents(nullptr);
-
-	signalHandlers.clear();
-
-	Auth::Save();
-	SaveProjectNow();
-	auth.reset();
-
-	delete extraBrowsers;
-
-	config_set_string(App()->GetUserConfig(), "BasicWindow", "DockState", saveState().toBase64().constData());
-
-#ifdef BROWSER_AVAILABLE
-	if (cef)
-		SaveExtraBrowserDocks();
-
-	ClearExtraBrowserDocks();
-#endif
-
-	OnEvent(OBS_FRONTEND_EVENT_SCRIPTING_SHUTDOWN);
+	if (shouldPromptForClose()) {
+		event->ignore();
+		restart = false;
 
-	disableSaving++;
+		if (!isClosePromptOpen) {
+			blog(LOG_INFO, "[OBSBasic] Prompting for close");
```

Comment:
Is there value in this (and the other) log message(s) using `INFO` level or is this defensive verbose logging to enable "remote debugging" on users' machines?